### PR TITLE
Refactor dockertested job to handle more deps better

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @jupierce 

PTAl

### CODE REVIEW

File: jobs/package-dockertested/Jenkinsfile [hunk 1]
```diff
@@ -21,23 +21,33 @@ node('openshift-build-1') {
...
-	    echo "rhel7next: docker-${next_docker} container-selinux-${next_cselinux} container-storage-setup-${next_cstorage} skopeo-${next_skopeo} atomic-${next_atomic}"
-	    test_docker = latestVersion('docker', 'dockertested')
-	    test_cselinux = latestVersion('container-selinux', 'dockertested')
-	    test_cstorage = latestVersion('container-storage-setup', 'dockertested')
-	    test_skopeo = latestVersion('skopeo', 'dockertested')
-	    test_atomic = latestVersion('atomic', 'dockertested')
-	    echo "dockertested: docker-${test_docker} container-selinux-${test_cselinux} container-storage-setup-${test_cstorage} skopeo-${test_skopeo} atomic-${test_atomic}"
-	    if ( next_docker == test_docker && next_cselinux == test_cselinux && next_cstorage == test_cstorage && next_skopeo == test_skopeo && next_atomic == test_atomic) {
+		def versions = [:]
+		for(package in packages) {
```
Comment (jupierce) [inline jobs/package-dockertested/Jenkinsfile]: Unless this is in a `@NonCPS` annotated method, this type of iteration throws an exception in our Jenkins version. You can use traditional `for ( int i = 0.... )` style or versions = some_new_noncps_method( packages ).

File: jobs/package-dockertested/Jenkinsfile [hunk 2]
```diff
@@ -21,23 +21,33 @@ node('openshift-build-1') {
...
+			def package_versions = [:]
+			package_versions['next'] = latestVersion(package, 'rhel7next*')
+			package_versions['current'] = latestVersion(package, 'dockertested')
+			versions[package] = package_versions
+		}
+
+		def sync_necessary = false
+		def table = new StringBuffer()
```
Comment (jupierce) [inline jobs/package-dockertested/Jenkinsfile]: StringBuffer is restricted, so I would just use `""` and append.

File: jobs/package-dockertested/Jenkinsfile [hunk 3]
```diff
@@ -21,23 +21,33 @@ node('openshift-build-1') {
...
+		table << 'Package'.center(30) << 'Current'.center(30) << 'RHEL 7 Next'.center(30) << '\n'
```
Comment (jupierce) [inline jobs/package-dockertested/Jenkinsfile]: .center, .padRight are also restricted. I can add them if really necessary, but can't we get by with `\t` tabbing for imperfect but legible output?

File: jobs/package-dockertested/Jenkinsfile [hunk 4]
```diff
@@ -21,23 +21,33 @@ node('openshift-build-1') {
...
+		for(package in versions) {
```
Comment (jupierce) [inline jobs/package-dockertested/Jenkinsfile]: Another `@NonCPS` candidate.

**Comment by stevekuznetsov** [comment]
@jupierce @adammhaile PTAL

**Comment by adammhaile** [comment]
lgtm

**Comment by stevekuznetsov** [comment]
Thanks, let's try this out..

**Pull request merged**